pkg/apis/v1beta1/nodepool.go

```diff
@@ -190,6 +190,12 @@ type NodePool struct {
 	Status NodePoolStatus `json:"status,omitempty"`
 }
 
+// We need to bump the NodePoolHashVersion under these condition when we make an update to the NodePool CRD
+// 1. A change to the default value to existing fields that are already hashed
```

Comment:
```suggestion
// 1. A field changes its default value for an existing field that is already hashed
```

nit: wording for parallelism with the other conditions here